src/components/Timeline/src/StepLabel.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import clsx from 'clsx';
+
+import StepIcon, { StepIconProps } from './StepIcon';
+import { TimelineComponent } from './types';
+
+export interface StepLabelProps extends TimelineComponent {
+  /**
+   * The label displayed in the step icon.
+   */
+  icon?: number;
+
+  /**
+   * The component to render in place of the StepIcon.
+   */
+  StepIconComponent?: React.ReactElement;
+
+  /**
+   * Props applied to the StepIcon element.
+   */
+  StepIconProps?: Partial<StepIconProps>;
+}
+
+/**
+ * Shows a label in a Step component within a Stepper component.
+ * @param props The properties of a StepLabel component.
+ * @constructor Constructs an instance of StepLabel.
+ */
+export const StepLabel: React.FC<StepLabelProps> = (props: StepLabelProps) => {
+  let StepIconComponent: React.ReactElement;
+  if (props.StepIconComponent === undefined) {
+    StepIconComponent = (
+      <StepIcon
+        active={props.active}
+        completed={props.completed}
+        disabled={props.disabled}
+        icon={props.icon}
+        {...props.StepIconProps}
+      />
+    );
```

Comment:
Generally for this use-case there is the React `defaultProps` setup, but there are some troubles with making `defaultProps` work with TypeScript afaik so not sure whether this is actually the preferred solution or not. Might be something to look into.

Reply:
I couldn't get `defaultProps` to work, because it kept overwriting my value with the `defaultProps` value.